### PR TITLE
Feat/web interface

### LINKED ISSUE #3: Create a web interface that allows users to recreate a PR
We already have the functions in the code for the CLI. We just need to make them WASM functions that can be reused in the client (javascript). 
We need to create an Oauth app that prompts users to sign in with github which gives us access to recreate the pr.

### PR DESCRIPTION
**Pull request by igbopharaoh**
WIP

### CODE REVIEW

File: client/fresheyes-web/README.md
Comment (Extheoisah) [inline client/fresheyes-web/README.md:1]: We need the README.md to provide guidance on how to run the frontend with the backend service

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+"use client";
+
+import React, { useMemo, useState } from "react";
+import { GitHubServiceClient } from "@/proto/fresheyes_pb_service";
+import { PullRequest } from "@/proto/fresheyes_pb";
+import Link from "next/link";
+import Modal from "../modal/Modal";
+
+const FormSection = () => {
+  const client = useMemo(() => new GitHubServiceClient("http://localhost:50051"), []);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx]: lets not hardcode the grpc url. It should be a constant read from config/env

Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx]: Can we not initialize the GRPC in the client component? Maybe do it outside in another function in a separate file that we can call anywhere. I think that way we won't need the `useMemo`

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    setLoading({ loader: true, modal: false });
+
+    const pr = new PullRequest();
+    pr.setOwner(owner.trim());
+    pr.setRepo(repo.trim());
+    pr.setPullNumber(pull_number);
+
+    const res = client.processPullRequest(pr, (error, result) => {
+      if (error) {
+        console.error(error);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx]: Lets store the error in a state and display the error in the UI

File: client/fresheyes-web/src/app/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import type { Metadata } from "next";
+import { Inter } from "next/font/google";
+import "./globals.css";
+import AuthProvider from "@/context/auth-provider";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata: Metadata = {
+  title: "Create Next App",
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/layout.tsx]:
```suggestion
  title: "Fresh-eyes",
```

File: client/fresheyes-web/src/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  description: "Generated by create next app",
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/layout.tsx]:
```suggestion
  description: "Fresh-eyes checks out a pull request (PR) branch, generates patches, and inserts GNU patch(1)-style diff comments, indicating where previous reviewers have left comments.",
```

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Looks good. Just a few comments

**Comment by igbopharaoh** [comment]
> Looks good. Just a few comments

Necessary corrections have been made to the comments

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 5]
```diff
@@ -44,8 +45,14 @@ const FormSection = () => {
 
     const res = client.processPullRequest(pr, (error, result) => {
       if (error) {
-        console.error(error);
-        setLoading({ loader: false, modal: false });
+        if (error.message.length === 0) {
+          setError("Opps an error occured you might want to check your setup");
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx]: typo `Oops!`.
Also, there's no setup required by the user. Let's just state that an error occurred. Please try again or contact support `info@bitcoindevs.xyz`

**Comment by Extheoisah** [comment]
Please rebase your branch on main. There are new changes to the server.

File: client/fresheyes-web/src/app/components/form/FormSection.tsx (hunk 5 above)
Comment (igbopharaoh) [reply client/fresheyes-web/src/app/components/form/FormSection.tsx]: Why I added this error message was when it returns a 404 error. The setup I meant here was ensuring the server was running

**Comment by Extheoisah** [comment]
The fresh-eyes bot files are back. I think you brought them back in your rebase

**Comment by igbopharaoh** [comment]
> The fresh-eyes bot files are back. I think you brought them back in your rebase

This is suprising but I've removed it again

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**